### PR TITLE
fix(rest): Fixed update Project API issue - unexpected changes in some fields like moderators ,contributors, etc

### PR DESCRIPTION
**Pull request by JaideepPalit**
> Fixed update Project API issue - unexpected changes in some fields like moderators ,contributors, etc
> * Which issue is this pull request belonging to and how is it solving it? (*#1057*)
> * Did you add or update any new dependencies that are required for your change? - No

### Checklist
Must:
- [X] All related issues are referenced in commit messages and in PR

Signed-off-by: Jaideep Palit <jaideep.palit@siemens.com>

### CODE REVIEW

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java [hunk 1]
```diff
@@ -154,6 +158,9 @@
     @NonNull
     private final Sw360ComponentService componentService;
 
+    @NonNull()
```
Comment (smrutis1) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java:161]: Please remove the empty parenthesis.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java [hunk 2]
```diff
@@ -617,10 +624,13 @@ public void downloadClearingReports(
     @RequestMapping(value = PROJECTS_URL + "/{id}", method = RequestMethod.PATCH)
     public ResponseEntity<Resource<Project>> patchProject(
             @PathVariable("id") String id,
-            @RequestBody Project updateProject) throws TException {
+            @RequestBody Map<String, Object> reqBodyMap) throws TException {
         User user = restControllerHelper.getSw360UserFromAuthentication();
         Project sw360Project = projectService.getProjectForUserById(id, user);
-        sw360Project = this.restControllerHelper.updateProject(sw360Project, updateProject);
+        ObjectMapper mapper = new ObjectMapper(); // jackson's objectmapper
```
Comment (smrutis1) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java:630]: would be good to remove the commented message.

**Review by smrutis1 — CHANGES_REQUESTED** [review]
Tested this PR. 

Observed one issue like after firing a REST patch request, the response contains `sw360:projects` tag, generally this should contain the self detail as per the documentaion. Unfortunately it is containing the project details of linked project.

**Comment by JaideepPalit** [comment]
Thanks for review and test.
Made changes as per review comments.

>Observed one issue like after firing a REST patch request, the response contains sw360:projects tag, generally this should contain the self detail as per the documentaion. Unfortunately it is containing the project details of linked project.

 - Fixed Documentation. `sw360:projects` is for linked Projects similar like in GetProjectByID.
The Main Doc contains project details and self link is provided in `"_links" ->"self"`

**Review by smrutis1 — APPROVED** [review]
Code looks good. Tested successfully

**Pull request merged**